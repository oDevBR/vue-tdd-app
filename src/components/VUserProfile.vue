
<script>
export default {
  name: 'UserProfile',
  props: {
    user: {
      type: Object,
      required: true,
      default: () => ({})
    }
  }
}
</script>

<template>
  <div class="user-profile" v-if="user.name">
    <el-card style="width: 50vw;" :body-style="{ padding: '0px', display: 'flex' }">
      <img :src="user.avatar_url" alt="" class="user-profile__avatar" />
      <div style="padding: 25px;">
        <span class="user-profile__name">{{ user.name }}</span>
        <p class="user-profile__location">{{ user.location }}</p>
      </div>
      <div class="user-profile__bio" style="padding: 2rem;">
        {{ user.bio }}
      </div>
    </el-card>
  </div>
</template>

<style>
  .user-profile {
    padding: 1em;
    display: flex;
    justify-content: center;
  }

  .user-profile__avatar {
    width: 100px;
    margin: 25px;
    box-shadow: 0 2px 12px 0 rgba(0,0,0,.1);
  }

  .user-profile__name {
    font-size: 1.5em;
  }

  .user-profile__location {
    color: #666;
  }
</style>