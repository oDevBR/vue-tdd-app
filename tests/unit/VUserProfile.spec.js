jest.mock('@/store/actions')
import { shallowMount, createLocalVue } from '@vue/test-utils'
import Element from 'element-ui'
import VUserProfile from '@/components/VUserProfile'
import user from './fixtures/user'

const localVue = createLocalVue()
localVue.use(Element)

describe('VUserProfile', () => {
    let props

    const build = () => {
        const wrapper = shallowMount(VUserProfile, {
            localVue,
            propsData: props
        })
        return {
            wrapper,
            avatar: () => wrapper.find('.user-profile__avatar'),
            name: () => wrapper.find('.user-profile__name'),
            bio: () => wrapper.find('.user-profile__bio'),
            location: () => wrapper.find('.user-profile__location')
        }
    }

    beforeEach(() => {
        props = {
            user
        }
    })

    it('should renders the component', () => {
        const { wrapper } = build()
        
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should renders main components', () => {
        const { avatar, name, bio, location } = build()

        expect(avatar().exists()).toBe(true)
        expect(avatar().attributes().src).toBe(props.user.avatar_url)

        expect(name().exists()).toBe(true)
        expect(name().text()).toBe(props.user.name)

        expect(bio().exists()).toBe(true)
        expect(bio().text()).toBe(props.user.bio)

        expect(location().exists()).toBe(true)
        expect(location().text()).toBe(props.user.location)
    })
})