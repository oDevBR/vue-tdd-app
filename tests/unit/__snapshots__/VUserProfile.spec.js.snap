// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserProfile should renders the component 1`] = `
<div class="user-profile">
  <el-card-stub bodystyle="[object Object]" style="width: 50vw;"><img src="https://avatars2.githubusercontent.com/u/33878228?v=4" alt="" class="user-profile__avatar">
    <div style="padding: 25px;"><span class="user-profile__name">Jo√£o Aquino</span>
      <p class="user-profile__location">Curitiba, PR, Brazil</p>
    </div>
    <div class="user-profile__bio" style="padding: 2rem;">
      Ninja Developer
    </div>
  </el-card-stub>
</div>
`;
